src/controller/CHIPDeviceController.h

```diff
@@ -116,6 +123,7 @@ class BLEDeviceConnectionParameters
     NodeId remoteDeviceId                   = 0;
     uint16_t discriminator                  = 0;
     uint32_t setupPINCode                   = 0;
+    std::string macAddress                  = "";
```

Comment:
since MAC addresses are of a known, fixed length, can we instead have a char array here in order to avoid trips to the heap?  this will address the API ask above, as well.